api/v1/sp.go

```diff
@@ -56,8 +74,19 @@ func handleApproveStorageProviders(c echo.Context) error {
 //	@Produce		json
 //	@Success		200	{object}	ResponseEnvelope{response=GetStorageProvidersResponse}
 //	@Router			/sp/suspend [post]
-func handleSuspendStorageProviders(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+func (a *apiV1) handleSuspendStorageProviders(c echo.Context) error {
+	var storageProviderIds Sps
+
+	err := json.NewDecoder(c.Request().Body).Decode(&storageProviderIds)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	var storageProviderResponse StorageProvidersResponse
+	a.db.Table("sps").Where("sp_id in (?)", storageProviderIds.SpId).UpdateColumn("tenant_sp_state", "suspended") //@jcace is active the right state
```

Comment:
It's only valid to suspend an SP thats in `active` state, other states are an error

Reply:
also use enum value ` db.TenantSpStateSuspended` please